filters/tags_filter.go

```diff
@@ -61,60 +44,100 @@ func (tn tagFiltersByNameAsc) Len() int           { return len(tn) }
 func (tn tagFiltersByNameAsc) Swap(i, j int)      { tn[i], tn[j] = tn[j], tn[i] }
 func (tn tagFiltersByNameAsc) Less(i, j int) bool { return bytes.Compare(tn[i].name, tn[j].name) < 0 }
 
+// TagsFilterOptions provide a set of tag filter options.
+type TagsFilterOptions struct {
+	// Name of the name tag.
+	NameTagName []byte
+
+	// Function to extract name and tags from an id.
+	NameAndTagsFn id.NameAndTagsFn
+
+	// Function to create a new sorted tag iterator from id tags.
+	SortedTagIteratorFn id.SortedTagIteratorFn
+}
+
 // tagsFilter contains a list of tag filters.
 type tagsFilter struct {
-	filters []tagFilter
-	iterFn  NewSortedTagIteratorFn
-	op      LogicalOp
+	nameFilter Filter
+	tagFilters []tagFilter
+	op         LogicalOp
```

Comment:
Do we intend to support more flexibility in how we construct a tagsFilter in the future? I'm thinking something like match (service:statsdex and (city:nyc or team:obs)).